### PR TITLE
Increase the test coverage of maps.py in test_maps.py

### PR DESCRIPTION
**Pull request by NhienPhan2909**
[x] Wrote test for feature
[ ] Added changes to CHANGELOG.md
[ ] Bumped version number (delete if unneeded)

**Changes proposed:**
Try to increase the test coverage of maps.py to as close to 100% as possible.

### CODE REVIEW

File: tests/test_maps.py [hunk 1]
```diff
@@ -195,3 +329,264 @@ def test_color_values_and_ids(states):
...
+            "coordinates": [[[0, 0], [1, 0], [1, 1], [0, 1], [0, 0]]]
+        },
+        "properties": {
+            "name": "Test Region"
+        },
+        "id": "test_id"
+    }
+    region = ds.Region(geojson_data)
+    # Call the geojson method with a new feature_id
+    updated_geojson = region.geojson("new_id")
```
Comment (adnanhemani) [inline tests/test_maps.py:550]: Isn't the new id named "test_id" above? Am I missing something?

**Review by adnanhemani — CHANGES_REQUESTED** [review]
Thank you for this - this is very needed.

But the unit tests are failing. Can you please take a look at the unit test results on this PR? It may help you to run the tests locally first to ensure that the tests aren't failing when you push to the PR. Please let me know if you have any questions - thank you!

File: tests/test_maps.py (hunk 1 above)
Comment (adnanhemani) [reply tests/test_maps.py:554]: Didn't understand the code earlier, makes sense now.

**Review by adnanhemani — APPROVED** [review]
After making the requested changes myself, this now looks good to me. But since I've done non-trivial work on this PR, I'd request @davidwagner to please review this first. I'll try to fill in the remaining untested code in a separate PR once this merges.

**Review by davidwagner — APPROVED** [review]

**Comment by davidwagner** [comment]
This looks good to me!  Thank you.  I appreciate all the tests and the bugfix in Map.read_geojson().  (Side note: I never noticed the semantics of Map.overlay(Map) before...)

**Comment by davidwagner** [comment]
@adnanhemani Merged.  Thank you for the review!

**Pull request merged**